src/Client/ClientFactory.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerAzure\Client;
+
+use GuzzleHttp\Client as GuzzleClient;
+use Psr\Http\Client\ClientInterface as HttpClientInterface;
+use SprykerAzure\Api\PullRequestApi\PullRequestApi;
+use SprykerAzure\Api\RepositoryApi\RepositoryApi;
+use SprykerAzure\Client\Builder\Plugin\ApiVersionPlugin;
+use SprykerAzure\Client\Builder\Plugin\BaseUrlPlugin;
+use SprykerAzure\Client\Builder\Plugin\JsonContentTypePlugin;
+use SprykerAzure\Client\Builder\Plugin\ResponseBodyJsonDeserializerPlugin;
+use SprykerAzure\Client\Builder\Plugin\ResponseThrowablePlugin;
+use SprykerAzure\Client\Builder\RequestBuilder;
+use SprykerAzure\Client\Builder\RequestBuilderInterface;
+use SprykerAzure\Client\Builder\ResponseDataBuilder;
+use SprykerAzure\Client\Builder\ResponseDataBuilderInterface;
+
+class ClientFactory
+{
+    /**
+     * @param \SprykerAzure\Client\Builder\RequestBuilderInterface|null $requestBuilder
+     * @param \SprykerAzure\Client\Builder\ResponseDataBuilderInterface|null $responseDataBuilder
+     * @param \Psr\Http\Client\ClientInterface|null $httpClient
+     *
+     * @return \SprykerAzure\Client\ClientInterface
+     */
+    public function createClient(
```

Comment:
Facade should not be behind the Factory. Facade calls factory and all the logic behind, not vise versa.
```suggestion
    //Usage
    $azureClient->addRequestPlugins([SomeReqPlugin()]);
    $azureClient->addResponsePlugins([SomeRespoPlugin()]);
    $response = $azureClient->createPullRequest($requestData);
```